limit.go

```diff
@@ -20,19 +23,6 @@ type Limit interface {
 	GetConnTotalLimit() int
 	// GetFDLimit returns the file descriptor limit.
 	GetFDLimit() int
-
-	// WithMemoryLimit creates a copy of this limit object, with memory limit adjusted to
```

Comment:
can we please keep the prototype constructors for limit in some form?

Reply:
These are not constructors. They’re methods on an interface, where they don’t belong.

Reply:
I haven't found any use for them. They conflate scaling and limit configuration. Disentangling those made this PR so invasive, but ultimately resulted in a cleaner structure.